backend/src/core/server.ts

```diff
@@ -49,6 +54,16 @@ export default class Server {
       socket.on(Event.DISCONNECT, socket => {
         logger.debug(`Socket ${socket.id} disconnected.`);
       });
+
+      socket.on(Event.CREATE_USERNAME, username => {
+        const extSocket = socket as ExtendedSocket;
+        if (username === "") {
+          extSocket.username = "Anonymous Adam";
```

Comment:
Is there some way we could make these unique? Maybe just doing Anonymous 1, Anonymous 2 ...

Reply:
Are we okay with having the room keep track of this global variable? We could also go w an NPM package that generates names as well. Or keep a list of animals and have it choose one out of there at random?

Reply:
Good question. I think actually doing the npm package that generates names would be good for now. [This one](https://www.npmjs.com/package/unique-names-generator) seems reasonably well maintained.

Reply:
Seems good because you can easily combine from their different dictionaries e.g. <colour>-<animal> => blue-squirrel